src/shared/components/modal.jsx

```diff
@@ -147,5 +146,16 @@ export default class Modal extends React.Component {
 
 Modal.propTypes = {
   visible: PropTypes.bool.isRequired,
-  onClose: PropTypes.func
+  children: PropTypes.element,
```

Comment:
as above, let's figure out whic of these props are actually required and which are not.

also, please move all these proptypes within the class definition

Reply:
Sure! Is `static propTypes` preferred? If so, why? I noticed both are used throughout the codebase.

Reply:
static proptypes is preffered whenever we have a component class, just because it's tidier -- it's clear immediately what the proptypes are for the class, as they are right next to the constructor. for functional components, the `X.PropTypes` notation is required.

TBH, i have no idea how this `Modal.propTypes` got through review in the first place :-~